src/blocks/gallery-masonry/edit.js

```diff
@@ -32,255 +32,233 @@ const masonryOptions = {
 	percentPosition: true,
 };
 
-class GalleryMasonryEdit extends Component {
-	constructor() {
-		super( ...arguments );
+const GalleryMasonryEdit = ( props ) => {
+	const {
+		attributes,
+		setAttributes,
+		wideControlsEnabled,
+		isSelected,
+		className,
+		editorSidebarOpened,
+		noticeUI,
+		pluginSidebarOpened,
+		publishSidebarOpened,
+	} = props;
 
-		this.onSelectImage = this.onSelectImage.bind( this );
-		this.onRemoveImage = this.onRemoveImage.bind( this );
-		this.onMove = this.onMove.bind( this );
-		this.onMoveForward = this.onMoveForward.bind( this );
-		this.onMoveBackward = this.onMoveBackward.bind( this );
-		this.setImageAttributes = this.setImageAttributes.bind( this );
-		this.replaceImage = this.replaceImage.bind( this );
+	const {
+		align,
+		animation,
+		captions,
+		gridSize,
+		gutter,
+		gutterMobile,
+		linkTo,
+		lightbox,
+		images,
+	} = attributes;
 
-		this.state = {
-			selectedImage: null,
-		};
-	}
+	const [ selectedImage, setSelectedImage ] = useState( null );
+	const [ , setCaptionSelected ] = useState( null );
```

Comment:
It seems weird that there is no variable. Is it normal?

Reply:
The linter wouldn't allow me to commit without removing the state value of this de-structure. It is valid syntax however I agree that it looks a bit odd. We could disable the linter rule for this specific thing as the state value wasn't being used, so with it present the linter was throwing an unused-var error

Reply:
Yeah, but my question is more : if there is no variable, there is no state, right? So, what is the purpose of the function?